README.md

```diff
@@ -136,122 +171,279 @@ Every resource in this repository is essential for the overall template setup. N
-  ```js
-    const OPERATIONS = ['publish', 'subscribe'];
+- `template` is a directory where you keep all the files that will be processed by the Generator using React. Only files with the `.js`, `.jsx` and `.cjs` extensions are taken and processed by the Generator. The rest are skipped.
+- `partials` is a directory where you keep reusable parts (called also components) of the templates. You can also put there helper functions to handle perform actions on AsyncAPI data inside components.
 
-    asyncapi.channelNames().forEach(channelName => {
-      const channel = asyncapi.channel(channelName);
+> **NOTE**: `partials` directory is recommended place to split reusable chunks/helpers. The reusable parts can be located both in the `template` folder as in another named folder. The only exception is the `hooks` folder, it is reserved for the Generator.
 
-      OPERATIONS.map((opName) => {
-        const op = channel[opName]();
-        if (!op) return;
-        
-        op.messages().forEach(m => {
-          //logic responsible for processing every message in every operation
-        });
-      });
-    });
-  ```
-- Example filter is used in [template/index.html](template/index.html) using Nunjucks syntax `{{ asyncapi | generateMermaidDiagram }}`
+> **NOTE**: If you are using React as a rendering engine, you should delete the given folders which are related to Nunjucks: `filters`, `template_njk`, `partials_njk`.
 
-### Hooks
+#### Template
 
-Hooks are functions called by the generator on a specific moment in the generation process. For more details, read about [hooks](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#hooks).
+Checkout [`template`](template) directory to see how different features of the generator are presented.
 
-#### Custom Template Hooks
+##### File component
 
-Check out [hooks/generateExtraFormats.js](hooks/generateExtraFormats.js) file to see an example of the hook. It is a hook that is invoked by the Generator after the template generation process is completed. Generator passes its context to hooks, which means you have access to data like `targetDir` that is a path to the directory where Template is generated, or `templateParams` with information about custom parameters passed by the user. This example hook provides optional features to the Template, like PNG/PDF/SVG generation, that the user decides on with custom parameters. That is not the only use case for the hook. There are more use cases like, for example, template cleanup operations after generation or modifications of the AsyncAPI document right before the generation.
+Each template, like in the [`template/index.js`](template/index.js) file should return in default export function as root the `<File>` component (or array of `<File>` components), which contains the necessary metadata for the Generator to render the file:
```

Comment:
Thanks, but:
- you cannot use class component because we don't check if default export is class or normal function.
- you can return `null`, `undefined` etc to not render template. See here https://github.com/asyncapi/generator-react-sdk/blob/d10601783a732be9299e5cabbaccf7fa0000da04/src/renderer/template.ts#L20 `undefined` is handled in generator side, not renderer.

For the generator to render a file with React certain conditions are required. 
1. The file should export a default function (example see the [`template/index.js`](template/index.js) file).
2. That function should return a `<File>` component as root component which contains the necessary metadata for the Generator to render the file. Returning `null`, `undefined` or another negative value forces the Generator to not create the file.

Will be good?

Reply:
Ahh, thought the component was possible because of the transpiler 😄 Yea sounds good 👍